.github/workflows/test.yaml

```
@@ -8,73 +8,69 @@ jobs:
         id: venv_cache
         with:
-          path: ~/.virtualenvs
-          key: poetry-${{ hashFiles('**/poetry.lock') }}
-          restore-keys: |
-            poetry-${{ hashFiles('**/poetry.lock') }}
+          enable-cache: true
+          cache-dependency-glob: "uv.lock"
 
-      - name: Configure Poetry
+      - name: Install venv
         run: |
-          python -m poetry config virtualenvs.in-project false
-          python -m poetry config virtualenvs.path ~/.virtualenvs
-
-      - name: Install libraries
-        run: python -m poetry install
-        if: steps.cache.outputs.cache-hit != 'true'
+          uv sync
 
       - name: Log in to DagsHub
         run: |
-          poetry run dagshub login --token ${{ secrets.DAGSHUB_TOKEN }}
+          uv run dagshub login --token ${{ secrets.DAGSHUB_TOKEN }}
+
+      - name: Install pip in UV
+        run: uv pip install pip
 
       - name: Download pretrained spacy libraries
-        run: poetry run spacy download en_core_web_sm
+        run: uv run -- spacy download en_core_web_sm
 
-      - name: check for data cache hit
-        uses: actions/cache@v3
-        if: env.caching == 'true'
-        id: data_cache
-        with:
-          path: ~/.fastai/data
-          key: 'fastai-test-data-v3'
+      # - name: check for data cache hit
+      #   uses: actions/cache@v3
+      #   if: env.caching == 'true'
+      #   id: data_cache
+      #   with:
+      #     path: ~/.fastai/data
+      #     key: 'fastai-test-data-v3'
 
-      - name: download data
-        if: env.download == 'true' && steps.cache.outputs.cache-hit != 'true'
-        run: |
-          wget https://raw.githubusercontent.com/fastai/docker-containers/master/fastai/tmp_scripts/download_testdata.py
-          ipython download_testdata.py
-          mkdir -p $HOME/.fastai/data
-          find $HOME/.fastai/archive/ -name "*.tgz" -exec tar -xzvf {} -C $HOME/.fastai/data \;
+      # - name: download data
+      #   if: env.download == 'true' && steps.cache.outputs.cache-hit != 'true'
+      #   run: |
+      #     wget https://raw.githubusercontent.com/fastai/docker-containers/master/fastai/tmp_scripts/download_testdata.py
+      #     ipython download_testdata.py
+      #     mkdir -p $HOME/.fastai/data
+      #     find $HOME/.fastai/archive/ -name "*.tgz" -exec tar -xzvf {} -C $HOME/.fastai/data \;
 
       - name: Pull data from DVC
         run: |
-          poetry run dvc remote modify origin --local auth basic
-          poetry run dvc remote modify origin --local user ${{ secrets.DAGSHUB_USERNAME }}
-          poetry run dvc remote modify origin --local password ${{ secrets.DAGSHUB_TOKEN }}
-          poetry run dvc pull
+          uv run dvc remote modify origin --local auth basic
+          uv run dvc remote modify origin --local user ${{ secrets.DAGSHUB_USERNAME }}
+          uv run dvc remote modify origin --local password ${{ secrets.DAGSHUB_TOKEN }}
+          uv run dvc pull
```

According to the [dagshub docs](https://dagshub.com/docs/feature_guide/dagshub_storage/#configuring-dvc) it seems this should use a key and secret instead.
```suggestion
          uv run dvc remote modify origin --local access_key_id ${{ secrets.DAGSHUB_TOKEN }}
          uv run dvc remote modify origin --local secret_access_key ${{ secrets.DAGSHUB_TOKEN }}
          uv run dvc pull
```